### ISSUE TITLE
Suggestion to make documentation far more usable

### ISSUE THREAD

**Issue by chuckdries** [Dogfalo/materialize#4746]
### Description
Many classes and related rules are not explicitly documented, merely implicitly used in examples scattered throughout other bits of code.

Classes like `.userView` and `.userView .background` are used in sample code, but as they are not written up as their own items, usability and discoverability issues arise: 

1. Because we don't document these classes explicity, they only appear in places we assume users might want them. If users want similar functionality but do not know of the class that exists to do whatever they need, they may write their own. This leads to duplication of effort and duplicate/unnecessary code in the final product.

2. Interaction between classes is unpredictable to users not experienced with the framework. For example: try placing the user card markup from the sideNav documentation into a `.card`, it totally breaks.

3. Many parts of the framework feel 'mysterious' and unpredictable, a quality of life issue that is not to be ignored.

### Repro Steps
Layout the steps to reproduce your issue.
Read any documentation page and note all the classes used within that are not explicitly explained.

Use this Codepen to demonstrate your issue. N/A

## Discussion  
This is more of a philosophical question to the community: Bootstrap documents literally every single class it includes. Do we want to follow in those footsteps? It's valid if we don't, but the framework as-is feels like it makes a lot of assumptions about how it is going to be used, and the lack of documentation makes it harder to work around those assumptions.

**Comment by chuckdries** [comment]
Another one I came upon is .collection-header vs .subheader. Subheaders are defined by google to be dividers for lists... which is basically what .collection-header does? Are subheaders only defined in a certain context? Why can't seem to use a collection in the sideNav? Documentation should make this functionality clear

**Comment by chuckdries** [comment]
Continuing the list of items where components are confusing in my opinion: today I wanted to add a search bar to my nav. I have two options:

1.  The method listed in the nav documentation takes up the entire nav bar, which is not what I want, but I can constrain it within a `.col` to make it work with other buttons, but that doesn't seem to be supported out of the box as I had to override a few `.row` styles and the height of the input itself to get it to show up correctly. 
2. The method used in the sideNav on the documentation site. This method uses `.search.` and `.search-wrapper` but isn't documented at all! I had to open my browser dev tools and inspect the documentation site, which is certainly not optimal.

Additionally, I wanted to display search results underneath: I could either go with `.collection` in keeping with the rest of the framework, or use the completely undocumented `.search-results`, which appears to be specific to sideNav.

I ended up stealing from both methods https://codepen.io/chuckdries/pen/yXaJLJ

I'm beginning to think that my issue may just be with sideNav lol.
